packages/kit/src/cli.js

```diff
@@ -183,7 +183,12 @@ prog
 			// these should take precedence, but not print conflict warnings
 			if (host) vite_config.preview.host = host;
 			if (https) vite_config.preview.https = https;
-			if (port) vite_config.preview.port = port;
+
+			if (port) {
+				vite_config.preview.port = port;
+			} else {
+				vite_config.preview.port ??= 3000;
```

Comment:
The default port of 3000 should probably be specified in the Vite plugin because the CLI will only be invoked when running the `svelte-kit` command, but if you're running with `vite`, Vitest, Storybook, etc. you'll end up with Vite's default port of `4173` in 3.0, which seems unexpected.

I don't understand why the code as written doesn't work. Maybe there's a Vite bug where the preview port isn't used from the inline config for some reason?

Reply:
Hmm. If I'm reading this correctly then all the plugins get applied after the inline config...

https://github.com/vitejs/vite/blob/d93ac8eca16534eb5474c19899bc130019b30a71/packages/vite/src/node/config.ts#L347

Reply:
I think Vite is behaving as expected: https://vitejs.dev/guide/api-plugin.html#config

I'm not quite sure what the best fix is here...

Reply:
A potential fix is to add an inline Vite plugin here to the `vite_config` using `enforce: 'post'` to force set the port (and other options). Or another way is to have the `sveltekit` plugin not set `port` (and other options) if they are already defined.

I think the latter may be better.